src/status.c

```diff
@@ -0,0 +1,104 @@
+#include <stdio.h> // Must be included before ebmusv2.h
+
+#include "ebmusv2.h"
+#include <assert.h>
+#include <commctrl.h>
+#include <stdarg.h>
+
+void format_status(int part, const char* format, ...) {
+	if (hwndStatus) {
+		va_list args;
+		va_start(args, format);
+		int size = vsnprintf(0, 0, format, args);
+		va_end(args);
+
+		char *buf = malloc(size + 1);
+		va_start(args, format);
+		vsprintf(buf, format, args);
+		va_end(args);
+		
+		SendMessage(hwndStatus, SB_SETTEXT, part, (LPARAM)buf);
+
+		free(buf);
+	}
+}
+
+static const char* notes[] = { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B" };
+
+void set_tracker_status(int p, BYTE *code) {
+	if (code[0] == 0x00) {
+		format_status(p, "End of pattern/subroutine");
+	} else if (code[0] < 0x80) {
+		if (code[1] < 0x80) {
+			format_status(p, "Note length: %d, velocity: %d/7, release: %d/15", code[0], (code[1] >> 4) & 0x7, code[1] & 0xF);
+		} else {
+			format_status(p, "Note length: %d", code[0]);
+		}
+	} else if (code[0] < 0xC8) {
+		BYTE note = (code[0] & 0x7F);
+		format_status(p, "Note %s%c", notes[note % 12], '1' + note / 12);
+	} else if (code[0] == 0xC8) {
+		format_status(p, "Hold");
+	} else if (code[0] == 0xC9) {
+		format_status(p, "Rest");
+	} else if (code[0] < 0xE0) {
+		format_status(p, "Percussion %d", code[0] - 0xC9);
+	} else {
+		switch (code[0]) {
+			case 0xE0: format_status(p, "Instrument %d", code[1]); break;
```

Comment:
To-do: special case for code[1] >= 0xCA, since that changes to a percussion instrument

Reply:
Added special note for values >= 0xCA